components/auth/auth-modal.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button } from "@/components/ui/button"
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog"
+
+interface AuthModalProps {
+  children: React.ReactNode
+}
+
+export function AuthModal({ children }: AuthModalProps) {
+  return (
+    <Dialog>
+      <DialogTrigger asChild>
+        {children}
+      </DialogTrigger>
```

Comment:
- no need to take children in props as this will be only a sign-in button and not anything else.